### DISCUSSION TITLE
Outdated dependencies

### DISCUSSION THREAD

**Discussion by odenix** [apple/pkl#97] [Q&A] [answered]
I've noticed that Pkl uses somewhat outdated tools and libraries.
For example, Gradle 7.5, Kotlin 1.7.10, and Clikt 3.5.1 were all released in 2022.

What's your policy for updating dependencies? Could Dependabot help?

**Comment by holzensp** [comment]
The Gradle update is a little cumbersome; 7.6 causes some tests to go flakey and 8.2 and beyond don't play nice with other dependencies. We need to retire JDK11 support (but we need a plan for that), but then we definitely want to turn on Dependabot.

**Comment by odenix** [comment]
It's strange that updating Gradle from 7.5 to 7.6 would cause flaky tests.
Anyway, how about updating other dependencies (e.g., Kotlin and Clikt) provided they're compatible with JDK11?
I'd be willing to send pull requests.

**Reply by holzensp** [reply, marked as answer]
It flummoxed me, also. Indeed, as long as we can stay JDK11 compatible for a _little_ bit longer, dependencies can be updated. Any such work would be incredibly appreciated!
